src/nexusrpc/handler/_common.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from datetime import timedelta
+from enum import Enum
+from typing import (
+    Generic,
+    Mapping,
+    Optional,
+    Sequence,
+)
+
+from nexusrpc import Link, OutputT
+
+
+class HandlerErrorType(Enum):
+    """Nexus handler error types.
+
+    See https://github.com/nexus-rpc/api/blob/main/SPEC.md#predefined-handler-errors
+    """
+
+    BAD_REQUEST = "BAD_REQUEST"
+    UNAUTHENTICATED = "UNAUTHENTICATED"
+    UNAUTHORIZED = "UNAUTHORIZED"
+    NOT_FOUND = "NOT_FOUND"
+    RESOURCE_EXHAUSTED = "RESOURCE_EXHAUSTED"
+    INTERNAL = "INTERNAL"
+    NOT_IMPLEMENTED = "NOT_IMPLEMENTED"
+    UNAVAILABLE = "UNAVAILABLE"
+    UPSTREAM_TIMEOUT = "UPSTREAM_TIMEOUT"
```

Comment:
Worth documenting each of these error types.

Reply:
Done